pkg/controller/dnszone/awsactuator.go

```diff
@@ -360,7 +360,11 @@ func (a *AWSActuator) Create() error {
 		return err
 	}
 
-	a.zoneID = hostedZone.Id
+	a.hostedZone = hostedZone
+	if err := a.ModifyStatus(); err != nil {
+		logger.WithError(err).Error("failed to populate DNSZone status")
+		return err
+	}
```

Comment:
This is snuck in the "just store entire aws dnszone" commit. Should it be in the "populate DNSZone status immediately after fetching zoneID (AWS)" commit instead?

Reply:
Yes, it came up while testing things and i must have squashed the wrong commits together while cleaning up the commits for submission.